lighthouse-core/audits/dobetterweb/no-document-write.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Audit a page to see if it's using document.write()
+ */
+
+'use strict';
+
+const Audit = require('../audit');
+const Formatter = require('../../formatters/formatter');
+
+class NoDocWriteAudit extends Audit {
+
+  /**
+   * @return {!AuditMeta}
+   */
+  static get meta() {
+    return {
+      category: 'JavaScript',
+      name: 'no-document-write',
+      description: 'Site does not use document.write()',
+      helpText: '',
```

Comment:
Add a suggestion like `Consider using <code>&lt;script async> to load scripts. <code>document.write()</code> is considered <a href="https://www.chromestatus.com/features/5718547946799104" target="_blank">harmful for performance</a>.`